esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -711,12 +766,25 @@ def __init__(
         self.method = method
         self.resolution = resolution
 
+        if src_mask is True:
+            src_mask = _get_mask(src_grid_cube)
+        elif src_mask is False:
+            src_mask = None
+        self.src_mask = src_mask
+        if tgt_mask is True:
+            tgt_mask = _get_mask(target_mesh_cube)
+        elif tgt_mask is False:
+            tgt_mask = None
+        self.tgt_mask = tgt_mask
+
```

Comment:
Since the logic is rather critical/sensitive (see parallel comments in `MeshToGridESMFRegridder`), it would be good to abstract this into a single function. Maybe you could make it part of `_get_mask()`? Or some other way.

Reply:
Addressed by https://github.com/SciTools-incubator/iris-esmf-regrid/pull/219/commits/ed880838184511b84efd63072f137ff25c4e0933